woocommerce/woocommerce
# WCA Test helper - add remote inbox notification staging importer

### Submission Review Guidelines:

-   I have followed the [WooCommerce Contributing Guidelines](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md) and the [WordPress Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/).
-   I have checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change.
-   I have reviewed my code for [security best practices](https://developer.wordpress.org/apis/security/).
-   Following the above guidelines will result in quick merges and clear and detailed feedback when appropriate.



### Changes proposed in this Pull Request:




This PR adds a new tool to the WCA Test Helper that helps import remote inbox notifications from staging or production for testing purposes. Since this tool cover functionalities from `Admin notes` tool, I've retired it.

`Import from staging` requires Automattic proxy connected.

- Import remote inbox notifications from https://staging.woocommerce.com or https://woocommerce.com
- Delete all 
- Delete a single notification
- Notification tester

 
![Screen Shot 2024-06-21 at 7 20 06 PM](https://github.com/woocommerce/woocommerce/assets/4723145/d61b3e4f-4f4e-40cd-8b22-058615c56466)

![Screen Shot 2024-06-21 at 7 20 29 PM](https://github.com/woocommerce/woocommerce/assets/4723145/bde8f3c7-2944-4be8-9907-b70cc292805b)



### How to test the changes in this Pull Request:



Using the [WooCommerce Testing Instructions Guide](https://github.com/woocommerce/woocommerce/wiki/Writing-high-quality-testing-instructions), include your detailed testing instructions:

1. Checkout this branch locally.
2. cd to `plugins/woocommerce-beta-tester` and run `pnpm run build:zip`
3. Create a new JN site with the latest WooCommerce
4. Upload and activate `woocommerce-beta-tester.zip`
5. Go to `Tools -> WCA Test Helper -> Remote Inbox Notifications`
6. Click on `Import from staging`
7. Confirm it throws an error.
8. Connect to Automattic Proxy.
9. Click on `Import from staging` again.
10. Confirm it works and notifications from https://staging.woocommerce.com/wp-json/wccom/inbox-notifications/1.0/notifications.json have been imported.





### Changelog entry







<details>

<summary>Changelog Entry Details</summary>

#### Significance




#### Type




#### Message 

Adds a new tool to the WCA Test Helper that helps import remote inbox notifications from staging or production for testing purposes

</details>

<details>

<summary>Changelog Entry Comment</summary>

#### Comment 

</details>
